src/lib.rs

```diff
@@ -56,11 +56,42 @@ impl Cli {
 
         let delete_file = run::generate_delete_file(tmp_dir.path())?;
 
-        run::start_shell(config, tmp_dir.path())?;
+        let mut shell = match run::start_shell(config, tmp_dir.path()) {
+            Ok(shell) => shell,
+            Err(err) => {
+                run::clean_up(
+                    delete_file,
+                    tmp_dir,
+                    self.worktree_branch.clone(),
+                    Vec::new(),
+                )?;
+                bail!("{}", err);
+            }
+        };
+
+        let subprocesses = run::start_subprocesses(config, tmp_dir.path());
```

Comment:
Should be fun before the shell otherwise foreground process' output will mix with the shell